tests/unit/test_client.py

```diff
@@ -138,3 +141,76 @@ async def test_CloudSQLClient_user_agent(
         assert client._user_agent == f"cloud-sql-python-connector/{version}+{driver}"
     # close client
     await client.close()
+
+
+async def test_cloud_sql_error_messages_get_metadata(
+    fake_credentials: Credentials,
+) -> None:
+    """
+    Test that Cloud SQL Admin API error messages are raised for _get_metadata.
+    """
+    # mock Cloud SQL Admin API calls with exceptions
+    client = CloudSQLClient(
+        sqladmin_api_endpoint="https://sqladmin.googleapis.com",
+        quota_project=None,
+        credentials=fake_credentials,
+    )
+    get_url = "https://sqladmin.googleapis.com/sql/v1beta4/projects/my-project/instances/my-instance/connectSettings"
+    resp_body = {
+        "error": {
+            "code": 403,
+            "message": "Cloud SQL Admin API has not been used in project 123456789 before or it is disabled",
+        }
+    }
+    with aioresponses() as mocked:
+        mocked.get(
+            get_url,
+            status=403,
+            payload=resp_body,
+            repeat=True,
+        )
+        try:
+            await client._get_metadata("my-project", "my-region", "my-instance")
+        except ClientResponseError as e:
+            assert e.status == 403
+            assert (
+                e.message
+                == "Cloud SQL Admin API has not been used in project 123456789 before or it is disabled"
+            )
```

Comment:
Will this test pass if `_get_metadata()` doesn't raise an exception? We want `_get_metadata()` to raise an exception. So should we do something like?:
```
with pytest.raises(ClientResponseError) as exc_info:
    assert exc_info.status == 403
    assert exc_info.message == "Cloud SQL Admin API has not been used in project 123456789 before or it is disabled"
```

Same for `test_cloud_sql_error_messages_get_ephemeral`

Reply:
Good catch! Updated to use pytest.raises 😄